app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  before_save :downcase_email
```

Comment:
- `before_save`: method này là gì em? mục đích dùng để làm gì?
- Ngoài method này, Rails cung cấp những method nào tương tự?

Reply:
+ before_save là 1 callback ạ, mục đích của method này là sẽ chuyển email thành ký tự in thường trước khi được lưu lại.
+ ngoài method này, rails còn có nhiều method khác như: before_validation, before_create, ....